controllers/module/kernelmapper.go

```diff
@@ -41,3 +65,50 @@ func (kernelMapper) FindMappingForKernel(mappings []ootov1alpha1.KernelMapping,
 
 	return nil, errors.New("no suitable mapping found")
 }
+
+func (k *kernelMapper) GetNodeOSConfig(node *v1.Node) *NodeOSConfig {
+	osConfig := NodeOSConfig{}
+
+	osConfigFieldsList := regexp.MustCompile("[.,-]").Split(node.Status.NodeInfo.KernelVersion, -1)
+
+	osConfig.fullKernelVersion = node.Status.NodeInfo.KernelVersion
+	osConfig.kernelVersionMMP = strings.Join(osConfigFieldsList[:kernelVersionPatchIdx+1], ".")
+	osConfig.kernelVersionMajor = osConfigFieldsList[kernelVersionMajorIdx]
+	osConfig.kernelVersionMinor = osConfigFieldsList[kernelVersionMinorIdx]
+	osConfig.kernelVersionPatch = osConfigFieldsList[kernelVersionPatchIdx]
+
+	if strings.Contains(node.Status.NodeInfo.OSImage, "Red Hat") {
```

Comment:
Is there another k8s distro that uses RHEL/RHCOS? If not, for OCP, won't client use the d/s operator instead? Do we really need to support Redhat u/s? Feels like a duplication effort with d/s don't you think?

Reply:
we would eventually also support Ubuntu and and other distributions. and currently i am treating this code as upstream  code. Once we have other distribution to test , we will expand this code

Reply:
I don't think we should expect u/s to be run on OCP. I think we should encourage clients to use the d/s project if they are using OCP. 

I am not aware of any other k8s distro (rather than OCP) that runs RHEL on the nodes so this is just misleading as we don't intend to keep support for OCP on this project (regarding all the security constrains etc).

@bthurber WDYT?

Reply:
As agreed, add a TODO that we will remove that code from upstream once we have it in downstream